### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'testEnvironmentOptions')

### ISSUE THREAD

**Issue by nzcodarnoc** [nrwl/nx#13518]
### Current Behavior

I'm experiencing this issue after an upgrade from Nx 13 to Nx 15.

Running React Native E2E tests fail with this error:

```sh
    TypeError: Cannot read properties of undefined (reading 'testEnvironmentOptions')
```

### Expected Behavior

Perhaps it is stating the obvious to say that I expect these commands to result in running E2E tests.

### Github Repo

https://github.com/nzcodarnoc/testEnvironmentOptionsDiagnostic

### Steps to Reproduce

```sh
npx create-nx-workspace@15.0.13 dummyWS
# ✔ Choose your style                     · integrated
# ✔ What to create in the new workspace   · react-native
# ✔ Application name                      · dummy
# ✔ Enable distributed caching to make your CI faster · No
yarn global add nx
cd dummy-ws
nx run-ios dummy
nx run dummy-e2e:test-ios:production
```

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.16.0
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx : 15.0.13
   @nrwl/angular : Not Found
   @nrwl/cypress : 15.0.13
   @nrwl/detox : 15.0.13
   @nrwl/devkit : 15.0.13
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.0.13
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 15.0.13
   @nrwl/js : 15.0.13
   @nrwl/linter : 15.0.13
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.0.13
   @nrwl/react-native : 15.0.13
   @nrwl/rollup : 15.0.13
   @nrwl/schematics : Not Found
   @nrwl/storybook : 15.0.13
   @nrwl/web : 15.0.13
   @nrwl/webpack : 15.0.13
   @nrwl/workspace : 15.0.13
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### Failure Logs

```shell
FAIL  src/app.spec.ts
  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'testEnvironmentOptions')

      10 |     super(config, context);
      11 |
    > 12 |     // Can be safely removed, if you are content with the default value (=300000ms)
         |     ^
      13 |     this.initTimeout = 300000;
      14 |
      15 |     // This takes care of generating status logs on a per-spec basis. By default, Jest only reports at file-level.

      at new NodeEnvironment (../../node_modules/jest-environment-node/build/index.js:96:49)
      at new DetoxCircusEnvironment (../../node_modules/detox/runners/jest-circus/environment.js:27:5)
      at new CustomDetoxEnvironment (environment.js:12:5)
      at async TestScheduler.scheduleTests (../../../../.config/yarn/global/node_modules/@jest/core/build/TestScheduler.js:333:13)
      at async runJest (../../../../.config/yarn/global/node_modules/@jest/core/build/runJest.js:404:19)
      at async _run10000 (../../../../.config/yarn/global/node_modules/@jest/core/build/cli/index.js:320:7)

14:23:21.014 detox[47960] ERROR: [cli.js] Command failed: jest --config jest.config.json --testNamePattern '^((?!:android:).)*$' --maxWorkers 1 e2e
```

### Additional Information

Running on MacOS Ventura 13.0.1 on an M1.

Output of `yarn global list`
```sh
yarn global v1.22.19
warning ../../package.json: No license field
info "detox@19.12.5" has binaries:
   - detox
info "detox-cli@19.0.0" has binaries:
   - detox
info "expo@42.0.4" has binaries:
   - expo
info "ios-deploy@1.11.4" has binaries:
   - ios-deploy
info "jest@28.1.1" has binaries:
   - jest
info "kill-port@2.0.1" has binaries:
   - kill-port
info "nx@15.0.13" has binaries:
   - nx
info "react-devtools@4.25.0" has binaries:
   - react-devtools
✨  Done in 0.93s.
```

**Comment by hcharley** [comment]
I'm also seeing this after switching from NPM to PNPM.

**Comment by AngelLoredoAlvarez** [comment]
I made a switch from Yarn to PNPM, and had a los of issues, being this one of them, I solved it installing 'jest-environment-node' dependency at the root of my monorepo, of course, I installed the version that match my jest version.

**Comment by SalahAdDin** [comment]
> I'm also seeing this after switching from NPM to PNPM.

How can we report this? The problem occurs clearly with `pnpm`.

**Comment by damianKokot** [comment]
Hey!
I've just realized that there was a [commit](https://github.com/facebook/jest/commit/9c00942817cc9abf281dfe12e60422a0b1182d34) with breaking change, that changes config passed into constructor, and from now o config that was passed should be nested as projectConfig inside passed object (check [these lines](https://github.com/facebook/jest/commit/9c00942817cc9abf281dfe12e60422a0b1182d34#diff-7e6b39bebf54c704e027085672211430b8241b080e103d997757bcea915d2041R28-R29))

After applying these changes and invoking constructor like so:
```javascript
class IntegrationEnvironment extends NodeEnvironment {
  constructor(projectConfig) {
    super({ projectConfig });   // Instead of super(projectConfig);
    this.testDatabaseName = '';
  }
  ...
}
```
Everything started working just fine!

Does it solve your issue?

**Comment by jonathanpmartins** [comment]
Stuck without solution here.

**Comment by xiongemi** [comment]
this is a jest options. i could not replicate this issue locally. maybe it is fixed for jest 29 and detox 20.9.1.
```
    "jest": "^29.4.1",
    "jest-circus": "^29.4.1",
    "jest-environment-jsdom": "^29.4.1",
    "jest-react-native": "18.0.0",
    "detox": "^20.9.0",
```

closing this issue for now.
